app/builders/application_builder.rb

```diff
@@ -0,0 +1,7 @@
+class ApplicationBuilder
+  def self.call(*args, **kwargs)
+    new(*args, **kwargs).call
```

Comment:
I've read some documentation but I'm still not sure I'm super clear on `*args`, `**args` and `**kwargs`

Prior to your refactor, `BulkSubscriberListEmailBuilder` was expecting `**args`, but it is passed keyword arguments, so I would have expected its `.call` method to have been 
` def self.call(**kwargs)` instead. Your new `ApplicationBuilder` does `new(*args, **kwargs).call` so is it fine that it doesn't have a `**args`?

Reply:
Ah, yes, the whole `**` v `*` thing can be confusing. Really we shouldn't need `**` here at all, but I didn't want to start modifying the constructors of all the classes. Here's a summary:

- `*anyvarname` is a way to collect any arguments passed to the method
- `*anyvarname` is a way to collect any keyword arguments (as in `key: value`) passed to the method
- &block is a way to collect any block passed to the method (not used here)

First thing to note is the naming is irrelevant. Second thing to note, by way of explanation, is that using all three of the above is one way to alias one method to another, like this:

```
# convoluted alias technique
def my_alias(*args, **kwargs, &block)
  real_method(*args, **kwargs, block)
end

# better alias technique
alias_method :my_alias, :real_method
```

We're not really doing that here, but hopefully that makes it clearer what's going on?

Might also help to say this is the same technique as we use elsewhere e.g. here: https://github.com/alphagov/content-publisher/blob/master/app/services/application_service.rb.

Reply:
> First thing to note is the naming is irrelevant

OK that makes it all much clearer! 

> Really we shouldn't need ** here at all, but I didn't want to start modifying the constructors of all the classes.

Yes makes sense. Thanks.